src/structures/paging/frame.rs

```diff
@@ -109,6 +110,20 @@ impl<S: PageSize> AddAssign<u64> for PhysFrame<S> {
     }
 }
 
+impl<S: PageSize> CheckedAdd<u64> for PhysFrame<S> {
+    type Output = Self;
+    #[inline]
+    fn checked_add(self, rhs: u64) -> Option<Self::Output> {
+        let phys_addr_rhs = rhs
+            .checked_mul(S::SIZE)
+            .and_then(|addr| PhysAddr::try_new(addr).ok())?;
```

Comment:
Let me know if this is overkill and you'd like me to opt for a lighter weight checked op by removing the PhysAddr validity check.